ipn/prefs.go

```diff
@@ -34,7 +36,7 @@ type Prefs struct {
 }
 
 // IsEmpty reports whether p is nil or pointing to a Prefs zero value.
-func (uc *Prefs) IsEmpty() bool { return uc == nil || *uc == Prefs{} }
+func (uc *Prefs) IsEmpty() bool { return uc == nil || uc.Equals(&Prefs{}) }
```

Comment:
This just got a lot more expensive.

Maybe the compiler can prove that the new value doesn't escape and doesn't need to be allocated, but Equals is very expensive (using ToBytes).

I'd just write this method out now, checking each field (with a test like the other two recently added... `git grep fieldsOf`).

Reply:
Done, with tests. That transitively made me do the same thing with controlclient.Persist, since it also has fields that require custom comparators.